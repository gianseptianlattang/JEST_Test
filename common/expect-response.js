const { matchers } = require('jest-json-schema');
expect.extend(matchers);

export const expectStatus = (response, expectedStatus) => {
  expect((response).status).toEqual(expectedStatus);
};

export const expectSchema = (response, expectedSchema) => {
  expect((response).body).toMatchSchema(expectedSchema);
};

export const expectObject = (response, expectedObject) => {
  expect((response).body).toMatchObject(expectedObject);
};

export const expectSnapshot = (response) => {
  expect((response).body).toMatchSnapshot();
};