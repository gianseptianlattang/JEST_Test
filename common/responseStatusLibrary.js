export default {
  statusOk: 200,
  statusCreated: 201,
  statusNoContent: 204,
  statusBadRequest: 400,
  statusUnauthorized: 401,
  statusForbidden: 403,
  statusNotFound: 404,
  statusToLarge: 413,
  statusUnprocessableEntity: 422,
  statusInternalServerError: 500,
};
