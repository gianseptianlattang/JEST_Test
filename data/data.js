const userNumber = Math.floor(Math.random() * 1000);

export const bodyRegister = () => {
  const register = {
    name: "Clark Kent",
    email: `superman${userNumber}@mailsac.com`,
    phone: "5555551234",
    password: "123456",
    password_confirmation: "123456",
  };
  return register;
};

export const bodyLogin = (email) => {
  const login = {
    email: email,
    password: "123456",
  };
  return login;
};

export const bodyForgotPassword = (email) => {
  const forgotPassword = {
    email: email,
    title: "Project Name",
    link: "https://example.com/",
  };
  return forgotPassword;
};

export const bodyResetPassword = (email, token) => {
  const forgotPassword = {
    password: "123456",
    password_confirmation: "123456",
    email: email,
    token: token,
  };
  return forgotPassword;
};
