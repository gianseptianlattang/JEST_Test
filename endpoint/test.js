const supertest = require("supertest");
require("dotenv").config();

const api = supertest("https://jsonplaceholder.typicode.com");

export const create = (body) =>
  api
    .post("/posts")
    .set("Content-Type", "application/json")
    .set("Accept", "application/json")
    .send(body);
