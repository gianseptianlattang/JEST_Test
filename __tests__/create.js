import * as endpoint from '../endpoint/test';
import * as er from '../common/expect-response';
import rs from '../common/responseStatusLibrary';
import data from '../data/create';

let body;
let response;

describe('Create New Data', () => {
    test(`As a User, I can Create New Data`, async () => {
        body = data;
        response = await endpoint.create(data);
        er.expectStatus(response, rs.statusCreated);
        er.expectObject(response, data);
        er.expectSnapshot(response);
    });
});
