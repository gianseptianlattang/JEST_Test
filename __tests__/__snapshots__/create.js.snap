// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create New Data As a User, I can Create New Data 1`] = `
Object {
  "body": "motorcycle",
  "id": 101,
  "title": "recommendation",
  "userId": "12",
}
`;
